backend/dataall/modules/s3_datasets/services/dataset_table_service.py

```diff
@@ -85,19 +99,49 @@ def preview(uri: str):
         with context.db_engine.scoped_session() as session:
             table: DatasetTable = DatasetTableRepository.get_dataset_table_by_uri(session, uri)
             dataset = DatasetRepository.get_dataset_by_uri(session, table.datasetUri)
+            env_group = None
+            resource_link_name = None
+            database_name = table.GlueDatabaseName
             if (
                 ConfidentialityClassification.get_confidentiality_level(dataset.confidentiality)
                 != ConfidentialityClassification.Unclassified.value
             ):
-                ResourcePolicyService.check_user_resource_permission(
+                resource_policy = ResourcePolicyService.check_user_resource_permission(
                     session=session,
                     username=context.username,
                     groups=context.groups,
                     resource_uri=table.tableUri,
                     permission_name=PREVIEW_DATASET_TABLE,
                 )
-            env = EnvironmentService.get_environment_by_uri(session, dataset.environmentUri)
-            return AthenaTableClient(env, table).get_table(dataset_uri=dataset.datasetUri)
+
+                if not (
+                    (dataset.SamlAdminGroupName in context.groups)
+                    or (dataset.stewards in context.groups)
+                    or (dataset.owner == context.username)
+                ):
+                    consumption_details = DatasetService._resolve_consumption_details_resource_uri(
+                        session, table.tableUri, resource_policy.principalId
+                    )
+                    if not consumption_details:
+                        raise exceptions.ResourceUnauthorized(
+                            username=context.username,
+                            action=PREVIEW_DATASET_TABLE,
+                            resource_uri=table.tableUri,
+                        )
+                    database_name = database_name + '_shared'
```

Comment:
Some of this logic could be part of `_resolve_consumption_details_resource_uri` and the interface implementation. For example, the `_shared` suffix should not be here, it is a data_sharing parameter not a dataset parameter

Reply:
removed as part of removing Preview Tab for Confidential Data